services/Donation.js

```diff
@@ -60,6 +60,87 @@ class DonationService {
       return false;
     }
   }
+
+  /**
+   * @description get total donations
+   */
+  static async getTotal(startDate, endDate) {
+    try {
+      const matchQuery = {};
+      if (startDate & endDate) {
+        matchQuery.createdAt = { $gte: startDate, $lte: endDate };
+      }
+
+      const totalDonations = await Donation.aggregate([
+        { $match: matchQuery },
+        { $group: { _id: null, totalAmount: { $sum: "$transaction.amount" } } }
+      ]);
+
+      if (totalDonations.length === 0) {
+        return { totalAmount: 0 };
+      }
+
+      return { totalAmount: totalDonations[0].totalAmount };
+    } catch (e) {
+      throw new Error("Internal Server Error");
+    }
+  }
+
+  /**
+   * @description get total donations and donors
+   */
+  static async getTotalDonorsAndAmount() {
+    try {
+      const donationStats = await Donation.aggregate([
+        {
+          $group: {
+            _id: null,
+            totalAmount: { $sum: "$transaction.amount" },
+            uniqueDonorsCount: { $addToSet: "$donor.email" }
+          }
+        }
+      ]);
+
+      const result = {
+        totalAmount:
+          donationStats.length > 0 ? donationStats[0].totalAmount : 0,
+        uniqueDonorsCount:
+          donationStats.length > 0
+            ? donationStats[0].uniqueDonorsCount.length
+            : 0
+      };
+
+      return result;
+    } catch (e) {
+      throw new Error("Internal Server Error");
+    }
+  }
+
+  /**
+   * @description get all donations
+   * @param {Number} page
+   * @param {Number} limit
+   */
+  static async getDonations(page, limit) {
+    try {
+      const startIndex = (page - 1) * limit;
+
+      // Count total donations (optional for pagination info)
+      const totalDonations = await Donation.countDocuments();
```

Comment:
This might be an expensive operation. I don't think we want to count the entire documents every time a call is made to this endpoint.

What we should be doing is passing with this function is have it pass in an argument called a `start` which acts as a cursor to where we want to begin the search in the DB. 

We use that **_start_** to do a search among the collection and when we find the document - we add that document plus the next page of documents as defined by the **_limit_**.

The illustration below might help you understand this better:

```
          start <-------- limit ---------->end
|     |      |     |     |     |    |     |     |     |     |     |     |     |   
 
```

That means the API payload response will be: 

```
{
  data: Array<Transaction>
  next: end //<--- this point to **end** from my illustration, the front-end will pass this when they make the next call
  /** whatever else metadata we need to add */
}

```

Let me know if this makes sense @Chiemezuo, we can talk more about it when we meet.